sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -117,8 +123,19 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
         case LongType => tryParseLong(field)
         case _: DecimalType => tryParseDecimal(field)
         case DoubleType => tryParseDouble(field)
-        case TimestampNTZType => tryParseTimestampNTZ(field)
-        case TimestampType => tryParseTimestamp(field)
+        case DateType => tryParseDateTime(field)
+        case TimestampNTZType =>
```

Comment:
Maybe we can fix it as below:

```suggestion
        case TimestampNTZType options.inferDate => tryParseDateTime(field)
        case TimestampNTZType => tryParseTimestampNTZ(field)
```